torch/onnx/utils.py

```diff
@@ -1660,6 +1661,50 @@ def _export(
     return torch_out
 
 
+@_beartype.beartype
+def _add_onnxscript_fn(
+    model_bytes: bytes, graph: _C.Graph, custom_opsets: Mapping[str, int]
+) -> bytes:
+    """Insert model-included custom onnx-script function into ModelProto"""
+
+    # TODO(titaiwang): remove this when onnx becomes dependency
+    try:
+        import onnx
+    except Exception:
+        raise errors.OnnxExporterError("Module onnx is not installed!")
+
+    onnx_function_list = list()
+    # Iterate graph nodes to insert only the included custom
+    # function_proto into model_proto
+    for node in graph.nodes():
+        if symbolic_helper._is_custom_domain(node):
+            domain_op = node.kind()
+            domain, opname = domain_op.split("::")
+            specified_version = custom_opsets.get(domain, 1)
+            onnx_function_group = registration.registry.get_function_group(domain_op)
+            if onnx_function_group is not None:
+                onnx_fn = onnx_function_group.get(specified_version)
+                if onnx_fn is not None:
+                    # TODO: to_function_proto is onnx-script API and can be annotated
+                    # after onnx-script is dependency
+                    onnx_function_list.append(onnx_fn.to_function_proto())  # type: ignore[attr-defined]
```

Comment:
Do we need to deduplicate this list?

Reply:
What is the current behavior if the same `onnx_fn` is seen twice? Do we append two of them in the list?

Reply:
Ahh I see

Reply:
Done. I used a set. LMK if there is a better approach on your mind.

Reply:
Since this is doing a sort of linking-step, it would be good to pin down the desired behavior: if an added function calls other onnx functions (transitively), should they be linked as well or not?

Reply:
IIRC, @BowenBao brought this up in the meeting, and I remember we don't support this yet. So no, this doesn't include function calling function behavior. I will comment this on this logic.